File: src/main/java/it/giaquinto/springberry/model/user/implementation/NormalUser.java
package it.giaquinto.springberry.model.user.implementation;

import it.giaquinto.springberry.model.security.SecurityLevel;

import java.time.LocalDateTime;

public final class NormalUser extends BaseUser {

    public NormalUser(
            final String name,
            final String surname,
            final LocalDateTime created
    ) {
        super(name, surname, created);
    }

    @Override
    public boolean equals(Object obj) {
        return super.equals(obj);
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public SecurityLevel levelOf() {
        return SecurityLevel.NORMAL;
    }
}

File: src/main/java/it/giaquinto/springberry/model/user/implementation/NormalUser.kt
package it.giaquinto.springberry.model.user.implementation

import it.giaquinto.springberry.model.security.SecurityLevel
import java.time.LocalDateTime

class NormalUser(
    name: String?,
    surname: String?,
    created: LocalDateTime?
) : BaseUser(name, surname, created) {
    override fun equals(obj: Any?): Boolean {
        return super.equals(obj)
    }

    override fun hashCode(): Int {
        return super.hashCode()
    }

    override fun levelOf(): SecurityLevel? {
        return SecurityLevel.NORMAL
    }
}